main.py

```
@@ -85,38 +85,92 @@ def run(self):
+        super().__init__()
+        self._asset_url = asset_url
+        self._dest_path = dest_path
+
+    def run(self):
+        try:
+            req = urllib.request.Request(
+                self._asset_url,
+                headers={"User-Agent": "JobDocs"},
+            )
+            cancelled = False
+            with urllib.request.urlopen(req, timeout=60) as resp:  # nosec B310
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)